crates/apollo-compiler/src/database/hir.rs

```diff
@@ -1154,6 +1154,65 @@ impl Value {
     pub fn is_null(&self) -> bool {
         matches!(self, Self::Null { .. })
     }
+
+    /// Returns an `i32` if the value is a number and can be represented as an i32.
+    #[must_use]
+    pub fn as_i32(&self) -> Option<i32> {
```

Comment:
`as_` methods are usually from borrowed to borrowed values, but we are probably ok here right? There is even a precedent in [`Duration::as_secs`](https://doc.rust-lang.org/core/time/struct.Duration.html#method.as_secs) in std 🤔